packages/web3-core-helpers/src/formatters.js

```diff
@@ -239,7 +239,14 @@ var outputTransactionFormatter = function (tx) {
         tx.transactionIndex = utils.hexToNumber(tx.transactionIndex);
     tx.nonce = utils.hexToNumber(tx.nonce);
     tx.gas = utils.hexToNumber(tx.gas);
-    tx.gasPrice = outputBigNumberFormatter(tx.gasPrice);
+    if (tx.gasPrice !== null)
+        tx.gasPrice = outputBigNumberFormatter(tx.gasPrice);
+    if (tx.maxFeePerGas)
+        tx.maxFeePerGas = outputBigNumberFormatter(tx.maxFeePerGas);
+    if (tx.maxPriorityFeePerGas)
+        tx.maxPriorityFeePerGas = outputBigNumberFormatter(tx.maxPriorityFeePerGas);
+    if (tx.type)
```

Comment:
@spacesailor24 Should we be inferring the `type` based on the presence of `maxFeePerGas` for legacy transactions? My thinking is if connected to a node that is not running the fork (pre-fork) then type will just not be there for legacy transactions.

Reply:
@corymsmith We can't infer it's existence purely off of the presence of `maxFeePerGas` because `type` will exist for [EIP2930](https://eips.ethereum.org/EIPS/eip-2930) as well as future `types`

Reply:
To clarify I didn't mean just `maxFeePerGas` but with some logic similar to what you have in `_handleTxType` in your PR

Reply:
I see, I don't find the extra checks necessary though, as `if (tx.type)` will cover if it's defined or `undefined`

Reply:
Yeah I was thinking `type` wasn't being returned at all from pre-fork nodes so was curious if we needed to add it, I was incorrect, its already there :)